doc/api/widgets.rst

```diff
@@ -0,0 +1,43 @@
+.. _widgets_label:
+
+Widgets
+=======
+
+.. currentmodule:: simpleline.render.widgets
+
+Widgets are basic units to render items on a :class:`screen <simpleline.render.screen.UIScreen>`.
+Widgets can wrap a common text (:class:`TextWidget`) or to create empty lines
+(:class:`SeparatorWidget`) or they could be more complex structures (:class:`CheckboxWidget`).
+New widget can be also created see :ref:`create_custom_widget_label`.
+
+These widgets should be used in :ref:`containers_label`.
+
+Widgets classes
+---------------
+
+.. automodule:: simpleline.render.widgets
+    :members:
+    :inherited-members:
+    :show-inheritance:
+
+.. _create_custom_widget_label:
+
+Creating custom widget
+----------------------
+
+To create a custom widget you should subclass the :class:`Widget` class. The most important method
+for creating customized widget is :meth:`Widget.render`. This method is responsible for presenting
+a user textual information in some form.
+
+If the new widget is the composition of other widgets the :meth:`Widget.draw` method should be
+called on every widget used. The :meth:`Widget.write` method if the input are strings.
```

Comment:
`method if the input are strings.` -> `method should be called if the input contains strings.`